libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/GutenbergEditorFragment.java

```diff
@@ -364,6 +367,55 @@ public static boolean contentContainsGutenbergBlocks(String postContent) {
         return (postContent != null && postContent.contains(GUTENBERG_BLOCK_START));
     }
 
+    public static String replaceMediaFileWithUrl(Context context, @NonNull String postContent,
+                                                 String localMediaId, MediaFile mediaFile) {
+        if (mediaFile != null) {
+            String remoteUrl = StringUtils.notNullStr(Utils.escapeQuotes(mediaFile.getFileURL()));
+            // TODO: replace the URL
+            if (!mediaFile.isVideo()) {
+                // replace gutenberg block id holder with serverMediaId, and url_holder with remoteUrl
+                String oldImgBlockHeader = String.format("<!-- wp:image {\"id\":%s} -->", localMediaId);
+                String newImgBlockHeader = String.format("<!-- wp:image {\"id\":%s} -->", mediaFile.getMediaId());
+                postContent = postContent.replace(oldImgBlockHeader, newImgBlockHeader);
+
+                // replace class wp-image-id with serverMediaId, and url_holder with remoteUrl
+                String oldImgClass = String.format("class=\"wp-image-%s\"", localMediaId);
+                String newImgClass = String.format("class=\"wp-image-%s\"", mediaFile.getMediaId());
+                postContent = postContent.replace(oldImgClass, newImgClass);
+
+                // let's first find this occurrence and keep note of the position, as we need to replace the
+                // immediate `src` value before
+                int iStartOfWpImageClassAttribute = postContent.indexOf(newImgClass);
+                if (iStartOfWpImageClassAttribute != -1) {
+                    // now search negatively, for the src attribute appearing right before
+                    int iStartOfImgTag = postContent.lastIndexOf("<img", iStartOfWpImageClassAttribute);
+                    if (iStartOfImgTag != -1) {
+                        Pattern p = Pattern.compile("<img[^>]*src=[\\\"']([^\\\"^']*)");
+                        Matcher m = p.matcher(postContent.substring(iStartOfImgTag));
+                        if (m.find()) {
+                            String src = m.group();
+                            int startIndex = src.indexOf("src=") + 5;
```

Comment:
Can we put magic number 5 in some variable so that we can remember what's the idea behind it ?

Reply:
👍 addressed in 6c5ec0ad2e

Reply:
er, re-wrote it in 924725edbc to a class constant so it's easier to read